src/bin/languageserver/mod.rs

```diff
@@ -107,13 +127,18 @@ struct Files {
 //
 // More information can be found here: https://github.com/hyperledger/solang/pull/1411
 
+struct GlobalCache {
```

Comment:
This struct deserves a doc comment.

Reply:
Added a doc comment that describes this struct

Reply:
"stores information" is not descriptive. Why do you have a global cache and not a per-file cache? This needs to be explained

Reply:
@chioni16 FYI, doc comments like that usually explain the higher level semantics of that struct. For example, why does it exist, how is supposed to be used, when is it useful, what pitfalls exists when using it. The question Sean suggested is a great example.

This helps other developers reading through the code base to get a much quicker understanding of how everything works together.

Reply:
Hello @seanyoung @xermicus @LucasSte !

I have tried to improve the doc comments as per your suggestions. The changes can be found here - https://github.com/hyperledger/solang/pull/1490/commits/1eadcad7c97611031eded5ba913d9ff976918d5c
Please let me know if something can be improved further. 

I really appreciate the suggestions and your time. 
Thanks!